kore/src/Kore/Step/Simplification/Application.hs

```diff
@@ -71,7 +71,7 @@ simplify predicate application = do
         (symbolConstructor symbol)
         (length childrenCrossProduct)
         (length result)
-    return result
+    return (OrPattern.toPattern result)
```

Comment:
If this `OrPattern` contains multiple branches, won't this have to be unpacked from a `Pattern` to an `OrPattern` again later?

Reply:
Yes, but it will be unpacked byt the `or` simplifier, which unpacks it "the right way". Also see the `TermLike.hs` answer below.